google/cloud/storage/client.py

```diff
@@ -528,6 +528,77 @@ def _put_resource(
             _target_object=_target_object,
         )
 
+    def _post_resource(
+        self,
+        path,
+        data,
+        query_params=None,
+        headers=None,
+        timeout=_DEFAULT_TIMEOUT,
+        retry=DEFAULT_RETRY,
```

Comment:
In my concurrent retry work I've been following a policy of defaulting to None on internal methods, esp. those related to POST which frequently are not idempotent. User-facing will have sensible defaults (often conditional ones). This is just a convention I've adopted and it's certainly open to disagreement.

Reply:
1b081c8f1e5a2f27eb4c7254adc09ead21356120